rust-runtime/smithy-types/src/retry.rs

```diff
@@ -68,3 +70,151 @@ pub enum RetryKind {
     /// The response associated with this variant should not be retried.
     NotRetryable,
 }
+
+#[non_exhaustive]
+#[derive(Eq, PartialEq, Debug, Clone, Copy)]
+pub enum RetryMode {
+    Standard,
+    Adaptive,
+}
+
+#[derive(Debug)]
+pub struct RetryModeErr(String);
```

Comment:
```suggestion
pub struct RetryModeParseErr(String);
```

to match the stdlib FromStr implementations